en/components/combo.md

```diff
@@ -0,0 +1,391 @@
+The result is as follows:
+<div class="sample-container loading" style="height: 540px;">
+    <iframe id="combo-features-sample" frameborder="0" seamless width="100%" height="100%" src="{environment:demosBaseUrl}/combo-features" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<div>
+    <button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="combo-features-sample" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+<div class="divider--half"></div>
+
+## Features
+
+### Data Binding
+
+Basic usage of `igx-combo` bound to a local data source, defining `valueKey` and `displayKey`:
+
+```html
+<igx-combo [data]="localData" [valueKey]="'ProductID'" [displayKey]="'ProductName'"></igx-combo>
+```
+
+> Note: If `displayKey` is omitted them `valueKey` entity will be used instead.
+
+
+Remote binding, defining `valueKey` and `displayKey`, and exposing `onDataPreLoad` that allows to load new chunk of remote data to the combo (see the sample above as a reference):
+
+```html
+<igx-combo [data]="remoteData | async" (onDataPreLoad)="dataLoading($event)" [valueKey]="'ProductID'" [displayKey]="'ProductName'" ></igx-combo>
+```
+
+```typesciprt
+public ngOnInit() {
+    this.remoteData = this.remoteService.remoteData;
+}
+
+public ngAfterViewInit() {
+    this.remoteService.getData(this.combo.virtualizationState, (data) => {
+        this.combo.totalItemCount = data.count;
+    });
+}
+
+public dataLoading(evt) {
+        if (this.prevRequest) {
+            this.prevRequest.unsubscribe();
+        }
+
+        this.prevRequest = this.remoteService.getData(this.combo.virtualizationState, () => {
+            this.cdr.detectChanges();
+            this.combo.triggerCheck();
+        });
+    }
+```
+
+> Note: In order to have combo with remote data, what you need is to have a service that retrives data chunks from a server. 
+What the combo exposes is a `virtualizationState` property that gives state of the combo - first index and the number of items that needs to be loaded.
+The service, should inform the combo for the total items that are on the server - using the `totalItemCount` property.
+
+### Value Binding
+
+If we want to use a two-way data-binding, we could just use `ngModule` like this:
+
+```html
+<igx-combo #combo [(ngModel)]="values"></igx-combo>
+```
+
+```typescript
+@ViewChild('combo', { read: IgxComboComponent }) public combo: IgxComboComponent;
+get values() {
+    return this.combo.selectedItems();
+}
+set values(newValues: Array<any>) {
+    this.combo.selectItems(newValues);
+}
+```
+
+<div class="divider--half"></div>
+
+### Filtering
+By default filtering in the combo is enabled. However you can disable it using the following code:
+
+```html
+<igx-combo [filaterable]="false"></igx-combo>
+```
+
+<div class="divider--half"></div>
+
+<div class="divider--half"></div>
+
+### Custom Values
+Enabling the custom values will add missing from the list, using the combo's interface.
```

Comment:
If the custom values are enabled, the missing item could be added using the UI of the combo.